File: car-reservation-frontend/src/types/reservation.ts
import {Car, CarImpl} from './car';
import {User, UserImpl} from './user';

export interface Reservation{
    id: number;
    creationDate: Date;
    startDate: Date;
    endDate: Date;
    user: User;
    car: Car;
}

export class ReservationImpl implements Reservation{

    
    public id: number;
    public creationDate: Date;
    public startDate: Date;
    public endDate: Date;
    public user: User;
    public car: Car;

    constructor(startDate? : Date, car? : Car){
        this.id = 0;
        this.creationDate = new Date();
        if(startDate){
            this.startDate = startDate;
            let endDate : Date = new Date(startDate);
            endDate.setMinutes(59);
            endDate.setSeconds(59);
            endDate.setMilliseconds(999);
            this.endDate = endDate;
        }
        else{
            this.startDate = new Date();
            this.endDate = new Date();
        }
        this.car = car ?? new CarImpl();
        this.user = new UserImpl();
    }

}

File: car-reservation-backend/src/main/java/bd/carreservation/model/Reservation.java

package bd.carreservation.model;

import java.time.LocalDateTime;

import bd.carreservation.constants.Constants;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity

@Table(name = Constants.RESERVATION_TABLE_NAME)
public class Reservation {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long id;

	@Column(columnDefinition = "timestamp", name = "creation_date")
	private LocalDateTime creationDate;

	@Column(columnDefinition = "timestamp", name = "start_date")
	private LocalDateTime startDate;

	@Column(columnDefinition = "timestamp", name = "end_date")
	private LocalDateTime endDate;

	@ManyToOne

	@JoinColumn(name = "user_id")
	private User user;

	@ManyToOne

	@JoinColumn(name = "car_id")
	private Car car;

}
